File: UISwitchObjC/UISwitchObjC/ViewController.m
//
//  ViewController.m
//  UISwitchObjC
//
//  Created by ChuangLan on 2019/12/13.
//  Copyright © 2019 QuentinZang. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
    UISwitch *uiSwitch = [[UISwitch alloc] initWithFrame:CGRectMake(130, 120, 0, 0)];
    uiSwitch.thumbTintColor = UIColor.greenColor;
    uiSwitch.onTintColor = UIColor.orangeColor;
    [uiSwitch setOn:YES animated:YES];
    [uiSwitch addTarget:self action:@selector(switchChanged:) forControlEvents:UIControlEventValueChanged];
    [self.view addSubview:uiSwitch];
    
}

- (void)switchChanged:(UISwitch *)sender {
    NSString *message = @"Turn on the switch.";
                     if (!sender.on) {
                         message = @"Turn off the switch.";
                     }
    NSLog(@"%@",message);
}

- (IBAction)switchChangedTap:(UISwitch *)sender {
    NSString *message = @"Turn on the switch.";
                     if (!sender.on) {
                         message = @"Turn off the switch.";
                     }
    NSLog(@"%@",message);
}

@end

File: UISwitchSwift/UISwitchSwift/ViewController.swift
//
//  ViewController.swift
//  UISwitchSwift
//
//  Created by ChuangLan on 2019/12/13.
//  Copyright © 2019 QuentinZang. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

     override func viewDidLoad() {
           super.viewDidLoad()
           // Do any additional setup after loading the view.
           let rect = CGRect(x: 130, y: 100, width: 0, height: 0);
           let uiSwitch = UISwitch(frame: rect)
           uiSwitch.tintColor = .red
           uiSwitch.thumbTintColor = .purple
           uiSwitch.onTintColor = .orange
           uiSwitch.setOn(true, animated: true)
           uiSwitch.addTarget(self, action: #selector(ViewController.switchChanged(_:)), for: .valueChanged)
           self.view.addSubview(uiSwitch)
       }

       @objc func switchChanged(_ uiSwitch:UISwitch) {
           var message = "Turn on the switch."
           if !uiSwitch.isOn {
               message = "Turn off the switch."
           }
               print(message)
       }


    @IBAction func switchChangeTap(_ sender: UISwitch) {
        var message = "Turn on the switch."
                  if !sender.isOn {
                      message = "Turn off the switch."
                  }
                      print(message)
    }
    
}

